bin/propolis-server/src/lib/migrate/destination.rs

```diff
@@ -321,6 +328,114 @@ impl<T: AsyncRead + AsyncWrite + Unpin + Send> DestinationProtocol<T> {
                 self.import_device(&target, &device, &migrate_ctx)?;
             }
         }
+        self.send_msg(codec::Message::Okay).await
+    }
+
+    // Get the guest time data from the source, make updates to it based on the
+    // new host, and write the data out to bhvye.
+    async fn time_data(&mut self) -> Result<(), MigrateError> {
+        // Read time data sent by the source and deserialize
+        let raw: String = match self.read_msg().await? {
+            codec::Message::Serialized(encoded) => encoded,
+            msg => {
+                error!(self.log(), "time data: unexpected message: {msg:?}");
+                return Err(MigrateError::UnexpectedMessage);
+            }
+        };
+        info!(self.log(), "VMM Time Data: {:?}", raw);
+        let mut deserializer = ron::Deserializer::from_str(&raw)
+            .map_err(codec::ProtocolError::from)?;
+        let deserializer =
+            &mut <dyn erased_serde::Deserializer>::erase(&mut deserializer);
+        let time_data_src: vmm::time::VmTimeData =
+            erased_serde::deserialize(deserializer).map_err(|e| {
+                MigrateError::TimeData(format!(
+                    "VMM Time Data deserialization error: {}",
+                    e
+                ))
+            })?;
```

Comment:
Since the destination type is known here, I think this can just be written as `let time_data_src: VmTimeData = ron::from_str(&raw)` (with an appropriate error mapping)--no need to do any of the type erasure/boxing.

Reply:
fixed in 505caa0